### PR TITLE
Fix the cleanup of local secret file

### PR DESCRIPTION
**Pull request by pditommaso**
This PR fixes the cleanup of secret files created by the local provider

### CODE REVIEW

**Review by marcodelapierre — APPROVED** [review]
I have added `deleteOnExit()` also for when the path is preexisting.  
This makes the behaviour retroactive, for tmp files with matching hash.

What do you think?

**Comment by pditommaso** [comment]
Umm, not sure anymore it should be deleted 😄

**Comment by pditommaso** [comment]
The point that file could be deleted during a by a run while a concurrent execution is running. 

I think the temp file is removed, then the file name should be unique i.e. using a random uuid

**Comment by marcodelapierre** [comment]
See now - added a random UUID to the filename.  
I did not use the session uniqueID, as it seemed an overshooting to expose the session object all the way to this class.

File: modules/nextflow/src/main/groovy/nextflow/secret/LocalSecretsProvider.groovy [hunk 1]
```diff
@@ -198,11 +198,13 @@ class LocalSecretsProvider implements SecretsProvider, Closeable {
         if( !secretsMap )
             return null
 
-        final name = ".nf-${CacheHelper.hasher(secretsMap.values()).hash()}.secrets"
+        final name = ".nf-${CacheHelper.hasher(secretsMap.values()).hash()}-${UUID.randomUUID().toString()}.secrets"
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/secret/LocalSecretsProvider.groovy:201]: Let's just use the random uuid

Comment (marcodelapierre) [reply modules/nextflow/src/main/groovy/nextflow/secret/LocalSecretsProvider.groovy]: done now

**Comment by marcodelapierre** [comment]
unrelated commits should be fine now, see changelog

**Comment by marcodelapierre** [comment]
(still showing extra commits in commit list, as I rebased to sign off one commit)

**Pull request merged**